source/digits_hits/src/GateDoseActor.cc

```diff
@@ -389,25 +392,30 @@ void GateDoseActor::UserSteppingActionInVoxel(const int index, const G4Step* ste
       double Volume = mDoseToWaterImage.GetVoxelVolume();
 
       // Other particles should be taken into account (Helium etc), but bug ? FIXME
-      if (PartName== "proton" || PartName== "e-" || PartName== "e+" || PartName== "deuteron") {
+      if (PartName== "proton" || PartName== "e-" || PartName== "e+" || PartName== "deuteron"|| PartName== "gamma") {
         //if (PartName != "O16[0.0]" && PartName != "alpha" && PartName != "Be7[0.0]" && PartName != "C12[0.0]"){
 
         DEDX = emcalc->ComputeTotalDEDX(Energy, PartName, material, cut);
         DEDX_Water = emcalc->ComputeTotalDEDX(Energy, PartName, "G4_WATER", cut);
-        doseToWater=edep/density/Volume/gray*(DEDX_Water/1.)/(DEDX/(density*e_SI));
+        doseToWater=edep/density/Volume*(DEDX_Water/1.)/(DEDX/(density*e_SI));
         if (DEDX_Water == 0 || DEDX == 0) doseToWater = 0.0; // to avoid inf or NaN
       }
-      else {
+      else { 
+      // FIX ME FOR OTHER PARTICLES
         DEDX = emcalc->ComputeTotalDEDX(100, "proton", material, cut);
         DEDX_Water = emcalc->ComputeTotalDEDX(100, "proton", "G4_WATER", cut);
-        doseToWater=edep/density/Volume/gray*(DEDX_Water/1.)/(DEDX/(density*e_SI));
+        doseToWater=edep/density/Volume*(DEDX_Water/1.)/(DEDX/(density*e_SI));
         if (DEDX_Water == 0 || DEDX == 0) doseToWater = 0.0; // to avoid inf or NaN
       }
 
       GateDebugMessage("Actor", 2,  "GateDoseActor -- UserSteppingActionInVoxel:\tdose to water = "
                        << G4BestUnit(doseToWater, "Dose to water")
                        << " rho = "
                        << G4BestUnit(density, "Volumic Mass")<< Gateendl );
+    
+   	// Convert to Gy AFTER debug message
+	dose = dose / gray;
```

Comment:
yes sorry for this. doseToWater !  3 Lines, 1 error... It is totally me... But most important. Do you agree with the adding of photon in "if" test ?

Then what is next ? I do another pull request after have fixed your remarks ? it is my first time on github